src/IOStream.cpp

```diff
@@ -160,11 +169,15 @@ size_t UDPServerStream::write(uint8_t const* buffer, size_t buffer_size)
         return buffer_size;
 
     ssize_t ret = sendto(m_fd, buffer, buffer_size, 0, &m_si_other, m_s_len);
-    if (ret == -1 && errno != EAGAIN && errno != ENOBUFS){
+    if (ret == -1) {
+        if (errno == EAGAIN && errno == ENOBUFS) {
+            return 0;
+        }
+        else if (m_ignore_econnrefused && errno == ECONNREFUSED) {
+            return 0;
+        }
```

Comment:
You don't get this on send()/sendto(). This is an ICMP response you receive from the remote host so you get it on the subsequent recv() call

Reply:
I thought the same, but it's not true. You get it on the first recvfrom/sendto that interacts with the socket after the reception of the ICMP message.

Reply:
Interesting. That's not what linux manpages says. Anyways... :+1:

Reply:
I have to admit that I had the same understanding than you ... but then you piqued my interest

From udp(7): All errors documented for socket(7) or ip(7) may be returned by a send **or receive** on a UDP socket.

Then lists ECONNREFUSED as a special udp-specific error.

Then, sendto: "Additional errors may be generated and returned from the underlying protocol modules; see their respective manual pages."

It seems to be there. It's just very hard to spot.

Reply:
Ah, there it is. I missed the "additional errors may be generated (,,,)" part in sendto(). Thanks for clarifying...